ee/codegen/src/generators/vellum-variable-value.ts

```diff
@@ -217,6 +218,56 @@ class ImageVellumValue extends AstNode {
   }
 }
 
+class FunctionCallVellumValue extends AstNode {
+  private value: FunctionCall;
+
+  public constructor(value: FunctionCall) {
+    super();
+    this.value = value;
+  }
+
+  public write(writer: Writer): void {
+    const arguments_ = [
+      python.methodArgument({
+        name: "arguments",
+        value: new Json(this.value.arguments),
+      }),
+      python.methodArgument({
+        name: "name",
+        value: python.TypeInstantiation.str(this.value.name),
+      }),
+    ];
+
+    if (!isNil(this.value.id)) {
+      arguments_.push(
+        python.methodArgument({
+          name: "id",
+          value: python.TypeInstantiation.str(this.value.id),
+        })
+      );
+    }
+
+    if (!isNil(this.value.state)) {
+      arguments_.push(
+        python.methodArgument({
+          name: "state",
+          value: python.TypeInstantiation.str(this.value.state),
+        })
+      );
+    }
```

Comment:
[nit] we should remove this - `state` is something we foresee removing from out function call types in the future and should always resolve to FULFILLED in the backend